internal/pkg/instance/instance.go

```diff
@@ -207,6 +244,97 @@ func (i *File) Update() error {
 	if n, err := file.Write(b); err != nil || n != len(b) {
 		return fmt.Errorf("failed to write instance file %s: %s", i.Path, err)
 	}
+
+	return file.Sync()
+}
+
+// MountNamespaces binds /proc/<pid>/ns directory into instance folder
+func (i *File) MountNamespaces() error {
+	path := filepath.Join(filepath.Dir(i.Path), "ns")
+
+	oldumask := syscall.Umask(0)
+	defer syscall.Umask(oldumask)
+
+	if err := os.Mkdir(path, 0755); err != nil {
+		return err
+	}
+
+	nspath, err := filepath.EvalSymlinks(path)
+	if err != nil {
+		return err
+	}
+
+	src := fmt.Sprintf("/proc/%d/ns", i.Pid)
+	if err := syscall.Mount(src, nspath, "", syscall.MS_BIND, ""); err != nil {
+		return fmt.Errorf("mounting %s in instance folder failed: %s", src, err)
+	}
+
+	return nil
+}
+
+// GetNamespaces fills instance namespaces path
+func (i *File) GetNamespaces(configNs []specs.LinuxNamespace) error {
```

Comment:
Just a suggestion, if a method is a getter shouldn't it return `([]specs.LinuxNamepsace, error)` rather than modifying parameter?
As a style suggestion: explicit `GetXXX()` is not a go-way, simple `XXX()` is commonly used

Reply:
Changed `GetNamespaces` for `UpdateNamespacesPath`